### PR TITLE
refactor network, handlers, block by ancestor system

### PR DESCRIPTION
**Pull request by SteelStrider**

### CODE REVIEW

File: node/node.py [hunk 1]
```diff
@@ -327,7 +327,8 @@ def form_private_key_reveal_transaction(self):
     def form_penalize_violators_transaction(self, conflicts):
         for conflict in conflicts:
```
[the remark this answers is not in the document]
Comment (SteelStrider) [reply node/node.py]: While leaving with a note. And some refactor of broadcast_block() not broadcast_block_out_of_timeslot() which are deprecated.

File: verification/block_acceptor.py [hunk 2]
```diff
@@ -32,10 +32,11 @@ def validate_prev_hashes_are_tops(self, prev_hashes):
             if not prev_hash in tops:
                 raise AcceptionException("Block refers to blocks which were not top blocks at the moment!")
 
-
     def validate_timeslot(self, block, current_block_number):
         for prev_hash in block.prev_hashes:
-            prev_hash_number = self.epoch.dag.get_block_number(prev_hash)
+            prev_hash_number = self.epoch.dag.try_get_block_number(prev_hash)
```
[the remark this answers is not in the document]
Comment (SteelStrider) [reply verification/block_acceptor.py]: I agree. reverted

File: node/node.py [hunk 3]
```diff
@@ -394,146 +395,140 @@ def get_allowed_signers_for_next_block(self, block):
...
+            else:  # PROCESS ORPHAN BLOCK
+                orphan_block_verifier = OrphanBlockAcceptor(self.epoch, self.blocks_buffer, self.logger)
+                if orphan_block_verifier.check_if_valid(signed_block.block):
+                    self.blocks_buffer.append(signed_block)
+                    self.logger.info("Orphan block add to buffer")
+                    # for every parent for received block
+                    for prev_hash in signed_block.block.prev_hashes:  # check received block ancestor
+                        if prev_hash not in self.dag.blocks_by_hash:  # check parent in local dag
+                            # if parent not exist in local DAG ask for parent
+                            self.network.direct_request_block_by_hash(self.node_id, node_id, prev_hash)
```
[the remark this answers is not in the document]
Comment (SteelStrider) [reply node/node.py]: Its only send request. 
If there is no block in the network, we simply will not receive response.

File: node/node.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (konchunas) [reply node/node.py]: Here you should not broadcast block at all. Block should be requested by node when it receives Conflict transaction

[the remark this answers is not in the document]
Comment (SteelStrider) [reply node/node.py]: Not sure about 29.... if not it will be next in todo list

**Comment by SteelStrider** [comment]
closed for refactor

**Pull request closed without merging**